server/file_handler.cc

```diff
@@ -0,0 +1,59 @@
+#include "file_handler.h"
+#include <vector>
+#include <string.h>
+#include <cstdio>
+#include "http.h"
+
+RequestHandler::Status StaticFileHandler::Init(const std::string& uri_prefix, const NginxConfig& config){
+	this->m_uri_prefix_ = uri_prefix;
+	//assume the config argument has something like
+	// root /path1
+	//check if such a line exists
+	auto root_config_tokens = config.find("root");
+	if(root_config_tokens.size() < 2){
+		printf("StaticfileHandler.Init: Invalid config");
+		return INVALID_CONFIG;
+	}
+	this->m_root_path_ = root_config_tokens[1];
+	return OK;
+}
+
+RequestHandler::Status StaticFileHandler::HandleRequest(const Request& request, Response* response){
+	std::string req_uri = request.uri();
+	std::size_t prefix_pos = req_uri.find(this->m_uri_prefix_);
+	if(prefix_pos == std::string::npos || prefix_pos != 0){
+		printf("StaticfileHandler.HandleRequest: Bad request");
+		response->SetStatus(Response::BAD_REQUEST);
+		response->AddHeader("Content-Type",http::mime_type::ContentTypeAsString(http::mime_type::CONTENT_TYPE_TEXT_HTML));
+		return BAD_REQUEST;
+	}
+
```

Comment:
I believe we don't actually have to validate the URI -- instead, the webserver should pass the appropriate URI to the appropriate handler. This means that this (type) of code should reside in the webserver.